### PR TITLE
Bump buildx and docker vendoring

### PR DESCRIPTION
**Pull request by chris-crone**
**What I did**
Bumped buildx in go.mod so that we can use a newer containerd. This removes the annoying error message on macOS arm64:
```
$ ./bin/docker 
ERRO[0000] failure getting variant                       error="getCPUInfo for OS darwin: not implemented"
```

### CODE REVIEW

**Comment by rumpl** [comment]
> macOS arm64

Apple Silicon you mean?

File: go.mod [hunk 1]
```diff
@@ -2,20 +2,6 @@ module github.com/docker/compose-cli
 
 go 1.15
 
-// the distribution version from ecs plugin is quite old and it breaks containerd
```
Comment (rumpl) [inline go.mod:5]: You got me, I thought all were removed, but you just moved them below! :D

Comment (chris-crone) [reply go.mod:5]: Yeah, I tried to make it cleaner :)

**Comment by chris-crone** [comment]
This is in draft because a lot was added to `go.sum` so I'd like to debug a bit

File: go.mod (hunk 1 above)
Comment (chris-crone) [reply go.mod:5]: I bumped docker to v20.10.0-rc2 so this should now be a bit cleaner.

File: local/compose/build.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func (s *composeService) build(ctx context.Context, project *types.Project, opts
 		return nil
 	}
 	const drivername = "default"
-	d, err := driver.GetDriver(ctx, drivername, nil, s.apiClient, nil, nil, "", nil, project.WorkingDir)
+	d, err := driver.GetDriver(ctx, drivername, nil, s.apiClient, nil, nil, nil, "", nil, project.WorkingDir)
```
Comment (chris-crone) [inline local/compose/build.go:108]: Note to reviewer: One actual code change required

**Review by gtardif — CHANGES_REQUESTED** [review]
Testing the removed warning, I don't see any output when `docker build`: 
```
$ ./bin/docker compose build --workdir tests/compose-e2e/volume-test/
The new 'docker compose' command is currently experimental. To provide feedback or request new features please open issues at https://github.com/docker/compose-cli
[+] Building 0.0s (0/2)                                         
$  
```

Same with `compose up`, all the build output is masked (after rmi the images)

**Comment by gtardif** [comment]
As a matter of fact, the build actually fails test added in https://github.com/docker/compose-cli/pull/1039

**Comment by chris-crone** [comment]
Hmm.. I see a truncated output which is weird. I didn't change anything related to the stdout streaming so not sure what's causing this.

**Comment by chris-crone** [comment]
Can you try with this patch:
```diff
diff --git a/local/compose/build.go b/local/compose/build.go
index a2d8050..c192ae8 100644
--- a/local/compose/build.go
+++ b/local/compose/build.go
@@ -122,6 +122,7 @@ func (s *composeService) build(ctx context.Context, project *types.Project, opts
 	progressCtx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	w := progress.NewPrinter(progressCtx, os.Stdout, "auto")
+	defer w.Wait()
 
 	// We rely on buildx "docker" builder integrated in docker engine, so don't need a DockerAPI here
 	_, err = build.Build(ctx, driverInfo, opts, nil, nil, w)

```

**Review by gtardif — APPROVED** [review]
Don't see issues with log being truncated, seems to something else with your local env @chris-crone (apple M1 ??)

**Comment by chris-crone** [comment]
I see this on Intel as well so have moved to draft :(

**Comment by chris-crone** [comment]
Found the issue, I have `BUILDX_NO_DEFAULT_LOAD=1` set which is stopping the images from being exported

**Review by rumpl — APPROVED** [review]

**Review by gtardif — APPROVED** [review]

**Comment by gtardif** [comment]
@chris-crone did that explain the logs you didn't see as well ?

**Comment by chris-crone** [comment]
> did that explain the logs you didn't see as well ?

For some of the logs yes. The rest can be explained by the missing export.

**Pull request merged**